### PR TITLE
Redis backend [WIP, Merge #12 first]

### PR DESCRIPTION
**Pull request by perrygeo**
Start of a Redis backend for tracking progress. Uses the pluggable backend from #12 

TODO
* [x] unit tests (~~still evaluating mocking options for redis~~ going with `mockredispy`)
* [x] live integration testing
* [x] benchmarking
* [x] docs
* [x] integration tests against redis mocks
* [x] command line interface to generalize to different backends

@dnomadb @vincentsarago: once I get unit tests in place, I'll ping you for review.

### CODE REVIEW

File: watchbot_progress/backends/redis.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        pipe.scard(self._parts_key(jobid))
+        meta, remaining = pipe.execute()
+
+        meta = self._decode_dict(meta)
+        try:
+            total = int(meta['total'])
+        except KeyError:
+            raise JobDoesNotExist('Job does not exist, run set_total first')
+
+        percent = (total - remaining) / total
```
Comment (vincentsarago) [inline watchbot_progress/backends/redis.py]: does this return only 0 or 1 in python 2.7 ?

Comment (perrygeo) [reply watchbot_progress/backends/redis.py]: It will always return a float because we're using `from __future__ import division`

Comment (vincentsarago) [reply watchbot_progress/backends/redis.py]: great didn't catch that 👌

**Comment by perrygeo** [comment]
@vincentsarago @dnomadb This is close, ready for review 👀 . Let me know if you have any questions.

I added two additional methods on the WatchbotProgress objects:

* `list_jobs` to list all jobids
* `list_pending_parts` to list incomplete partids

And refactored the CLI to use these methods.

The CLI can be used with dynamodb or redis by using `--database arn:to:dynamo:table` or `--database redis://localhost:6379` for example.

**Comment by perrygeo** [comment]
Merging this with the intention of getting a `0.5.0a1` alpha release out today.

**Pull request merged**